### PR TITLE
Initialize API

### PR DESCRIPTION
**Pull request by vitzlo**
Implements functionality to:
- Get the data for a specific Pokemon based on its name
- Get all the Pokemon names that fit the search query (i.e. query is a substring of the name)

This is what I used to spot check that it worked:
```jsx
    const [response, setResponse] = useState({});
    useEffect(() => {
        findPokemon("pichu").then(
            (pokemon) => {
                setResponse(pokemon);
            }
        );
    }, []);

    return <h1>{JSON.stringify(response)}</h1>
```

I replaced `findPokemon()` with a `searchPokemonBySubstring()` call to test as well. Any suggestions on what to include in the pruned data are welcome. 

[Documentation for find()](https://pokeapi.co/docs/v2#pokemon)
[API playground](https://pokeapi.co/)

### CODE REVIEW

File: src/Utils/client.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/////////////
+// HELPERS //
+/////////////
+
+/////////////
+// PRUNING // (since we don't want thaaaat much data)
+/////////////
+
+const pruneTypes = (typesData) => {
+  return typesData.map((t) => t.type.name);
```
Comment (vitzlo) [inline src/Utils/client.js:14]: we could potentially have a map of type names to colors to style info boxes, but for now names just seemed like the only relevant data for our purposes

File: src/Utils/client.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+// GET: list of Pokemon names (not full Pokemon data)
+// not case-sensitive
+export const searchPokemonBySubstring = async (substring) => {
+  const response = await axios.get(
+    `${API_BASE}/pokemon?limit=${POKEMON_TOTAL}`
+  ); // all pokemon
+
+  const allPokemon = response.data.results;
+  const filteredPokemon = allPokemon
+    .filter((pokemon) => pokemon.name.includes(substring.toLowerCase())) // has search query in name
```
Comment (vitzlo) [inline src/Utils/client.js:64]: is this enough for "how should searching actually work", or do we want subsequence, which i can implement fairly quickly

File: src/Utils/client.js (hunk 1 above)
Comment (xiushak) [reply src/Utils/client.js:14]: that could be a fun feature to add when we finish everything else

File: src/Utils/client.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  return typesData.map((t) => t.type.name);
+};
+
+const pruneSprites = (spritesData) => {
+  const prunedData = {};
+
+  // this field seemed suspect so this is a cautious measure
+  if (
+    spritesData["other"] !== undefined &&
+    spritesData["other"]["official-artwork"] !== undefined
```
Comment (xiushak) [inline src/Utils/client.js]: can just do if (!spritesData["other] && !......) instead of !== undefined

File: src/Utils/client.js (hunk 2 above)
Comment (xiushak) [reply src/Utils/client.js:64]: most pokemon are 1 word names so i think this is good enough

File: src/Utils/client.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  return prunedData;
+};
+
+/////////////
+// EXPORTS //
+/////////////
+
+export const findPokemon = async (name) => {
+  const response = await axios.get(`${API_BASE}/pokemon/${name}`);
+  if (response.data === undefined) {
```
Comment (xiushak) [inline src/Utils/client.js]: same here, just !response.data

File: src/Utils/client.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+};
+
+// GET: list of Pokemon names (not full Pokemon data)
+// not case-sensitive
+export const searchPokemonBySubstring = async (substring) => {
+  const response = await axios.get(
+    `${API_BASE}/pokemon?limit=${POKEMON_TOTAL}`
+  ); // all pokemon
+
+  const allPokemon = response.data.results;
```
Comment (xiushak) [inline src/Utils/client.js:62]: would it be better to map it to names first here before doing the filter?

**Review by xiushak — CHANGES_REQUESTED** [review]
looks good, just some minor changes!

File: src/Utils/client.js (hunk 3 above)
Comment (vitzlo) [reply src/Utils/client.js]: `spritesData["other"]` works yup

File: src/Utils/client.js (hunk 5 above)
Comment (vitzlo) [reply src/Utils/client.js:62]: runtime-wise filtering as soon as possible is quicker overall but accessing `pokemon.name` twice is kinda scuffed when we could do it once

i'm okay with changing it i just wanted to bring up that for most queries we're essentially cutting our processing time in half by mapping later i think, i'll wait for response

**Review by xiushak — APPROVED** [review]
lgtm

**Pull request merged**